CMake/OpenAccHelper.cmake

```diff
@@ -68,7 +69,7 @@ if(CORENRN_ENABLE_GPU)
     GLOBAL
     PROPERTY
       CORENEURON_LIB_LINK_FLAGS
-      "${PGI_ACC_FLAGS} -rdynamic -lrt -Wl,--whole-archive -L${CMAKE_HOST_SYSTEM_PROCESSOR} -lcorenrnmech -L${CMAKE_INSTALL_PREFIX}/lib -lcoreneuron -lcudacoreneuron -Wl,--no-whole-archive ${CUDA_cudart_static_LIBRARY}"
+      "${PGI_ACC_FLAGS} -rdynamic -lrt -Wl,--whole-archive -L${CMAKE_HOST_SYSTEM_PROCESSOR} -lcorenrnmech -L${CMAKE_INSTALL_PREFIX}/lib -lcoreneuron -lcudacoreneuron -lcoreneuron -Wl,--no-whole-archive ${CUDA_cudart_static_LIBRARY}"
```

Comment:
I see `-lcoreneuron` twice. Expected?

Reply:
Yes, although I haven't actually tested the `nrnivmodl` workflow (only `nrnivmodl-core`) where IIUC this is used -- the CI will do that for us...

It's a crude workaround (commented elsewhere) for circular dependencies between `libcoreneuron.a` and `libcudacoreneuron.a` introduced by this PR. I didn't immediately find a better **simple** way.

IMO the better solution would be for us to produce a single library instead of these two separate ones. I don't think they are really logically separate libraries, the separation is just a consequence of how things are handled in CMake.

Alternatively we could add a third library...
```
libcoreneuron.a (C++) <depends on> libcudacoreneuron.a (CUDA) <depends on> libcorecoreneuron.a (C++)
```
but I fear this will just get messier without much real benefit.

Reply:
#609 made it easy to merge `libcoreneuron` and `libcudacoreneron`, so I did that instead. Much neater.